spring-cloud-stream-core-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc

```diff
@@ -1519,6 +1519,22 @@ This can be customized on the binding, either by setting a SpEL expression to be
 
 Additional properties can be configured for more advanced scenarios, as described in the following section.
 
+===== Accessing application context for PartitionKeyExtractorClass
+
+In cases where 'PartitionKeyExtractorClass' needs to access the underlying application context (to resolve application properties etc.,) the following can be done:
+
+You can configure your strategy implementation as a @Bean in the application context with a bean name of the fully qualified class name, that bean will be fully initialized by Spring and used by the binder.
```

Comment:
We should explicitly mention `PartitionKeyExtractorClass ` instead of the more generic 'strategy'.